activerecord/test/cases/inheritance_test.rb

```diff
@@ -315,6 +315,13 @@ def test_new_without_storing_full_sti_class
     end
   end
 
+  def test_new_when_passing_class_to_type_when_store_attributes_are_present_on_sti_class
+    without_store_full_sti_class do
+      item = Company.new(type: Firm)
```

Comment:
Never mind, we've never supported assigning a class to a string field. Passing it to `where` is supported, but is deprecated in Rails 5 and will be removed in 5.1